scripts/revng-check-conventions

```diff
@@ -100,134 +109,244 @@ while [[ $# > 0 ]]; do
+        cmake-format --check "${CMAKE_FILES[@]}" -l error
+    fi
+}
+
+# Run shellcheck + check for set -euo pipefail
+function run_bash_check() {
+    readarray -t BASH_FILES < <($bash_files)
+    if [[ ${#BASH_FILES[@]} -eq 0 ]]; then return; fi
+    shellcheck "${BASH_FILES[@]}"
+
+    local no_set_files=()
+    for FILE in "${BASH_FILES[@]}"; do
+        if ! head -n 10 "$FILE" | grep -qFe 'set -euo pipefail'; then
+            no_set_files+=("$FILE")
+        fi
+    done
+    if [[ ${#no_set_files[@]} -gt 0 ]]; then
+        echo "There are script files without 'set -euo pipefail'"
+        printf '%s\n' "${no_set_files[@]}"
     fi
 }
 
+# Check that all files have a license header
+function run_license_check() {
+    readarray -t ALL_FILES < <($GIT_LS_FILES | \
+        grep -Ev \
+            -e '\.txt$' \
+            -e '\.md$' \
+            -e '\.rst$' \
+            -e '.gitignore' \
+            -e 'LICENSE.*' \
+            -e 'pyproject.toml' \
+            -e 'clang-format-style-file' \
+            -e '\.dot$' \
+            -e 'Doxyfile.in'
+    )
+
+    local no_license_files=()
+    for FILE in "${ALL_FILES[@]}"; do
+        if ! head -n 10 "$FILE" | grep -qF 'See LICENSE.md for details'; then
+            no_license_files+=("$FILE")
+        fi
+    done
+
+    if [[ ${#no_license_files[@]} -gt 0 ]]; then
+        echo "There are files without a license header:"
+        printf '%s\n' "${no_license_files[@]}"
+    fi
+}
+
+
 # Run revng-specific checks on files
 function run_revng_checks() {
+    if [[ ${#FILES[@]} -eq 0 ]]; then return; fi
+    GREP="grep -Hn --color=always"
+
     # Check for lines longer than 80 columns
-    $GREP -E '^.{81,}$' $FILES | cat
+    $GREP -E '^.{81,}$' "${FILES[@]}" | cat_header "There are lines longer than 80 characters:"
 
     # Things should never match
     for REGEXP in '\(--> 0\)' ';;' '^\s*->.*;$' 'Twine [^&]'; do
-        $GREP "$REGEXP" $FILES | cat
-    done
+        $GREP "$REGEXP" "${FILES[@]}"
+    done | cat_header "There are things that should not match:"
 
     # Things should never match (except in support.c)
-    FILTERED_FILES="$(echo $FILES | tr ' ' '\n' | grep -v \
-        -e '^runtime/support\.c$' \
-        -e '^lib/Support/Assert\.cpp$' \
-        )"
+    readarray -t FILTERED_FILES < <(printf '%s\n' "${FILES[@]}" | \
+        grep -v -e '^runtime/support\.c$' -e '^lib/Support/Assert\.cpp$')
 
     for REGEXP in '\babort(' '\bassert(' 'assert(false' 'llvm_unreachable'; do
-        $GREP "$REGEXP" $FILTERED_FILES | cat
-    done
+        $GREP "$REGEXP" "${FILTERED_FILES[@]}"
+    done | cat_header "Assert code found where it should not be:"
 
     # Things should never be at the end of a line
     for REGEXP in '::' '<' 'RegisterPass.*>' '} else' '\bopt\b.*>'; do
-        $GREP "$REGEXP\$" $FILES | cat
-    done
+        $GREP "$REGEXP\$" "${FILES[@]}"
+    done | cat_header "Things that shouldn't be at the end of a line:"
```

Comment:
Ibidem. All so drop "things".

Reply:
Reworded all headers, they should sound better now.